src/lib/transaction_snark/test/transaction_union/transaction_union.ml

```diff
@@ -937,9 +927,75 @@ let%test_module "Transaction union tests" =
               assert (
                 Balance.equal fee_payer_account.balance
                   expected_fee_payer_balance ) ;
-              assert (Option.is_none source_account) ;
               assert (Option.is_some receiver_account) ) )
 
+    let%test_unit "Payment to and from same key" =
```

Comment:
I'd gladly see these tests moved to `transaction_snark/tests` directory. Situation where we have both inline tests and separate test directory for a library does not support comprehensive testing at all.

Reply:
I'm not sure I understand what you are suggesting. This is in the `transaction_snark/test` directory, under the `transaction_union` tests.

Reply:
Yeah… I wrongly thought it was located elsewhere. Sorry for the noise :sweat_smile: